Fix concretisation issues with python packages

pramodk opened the pull request:
* allow node-js to concretise with python3
* fix py-ipython package as latest develop has fixed concretisation issues
* fix py-bluepymm for python > 3.4
* add new version of py-markdown compatible unto python 3.7

pramodk committed:
Fix concretisation issues with python packages for deployment

ferdonline commented on var/spack/repos/builtin/packages/py-ipython/package.py:
```diff
@@ -39,12 +39,8 @@ class PyIpython(PythonPackage):
-    depends_on('py-backports-shutil-get-terminal-size', type=('build', 'run'))
-    depends_on('py-pathlib2',                   type=('build', 'run'))
+    depends_on('py-backports-shutil-get-terminal-size', type=('build', 'run'), when="^python@:3.2")  # noqa
+    depends_on('py-pathlib2', type=('build', 'run'), when="^python@:3.3")
```
> `when="^python@:3.2"`
> `when="^python@:3.3"`

if I remember well this only works when we append .99 as patch version, like you have for node.js. Or is there I understood wrong?

ferdonline replied:
I believe neurodamus needs mpi by default, unless we disable mpi in the whole hdf5 stack.

pramodk replied:
> > `when="^python@:3.2"`
> > `when="^python@:3.3"`
> 
> if I remember well this only works when we append .99 as patch version, like you have for node.js. Or is there I understood wrong?

I don't think so..that just version specifier..it says `from_version:to_version`. People add `:3.1.99` instead of `:3.2` because in later case you `include` 3.2 as well.

pramodk replied:
> I believe neurodamus needs mpi by default, unless we disable mpi in the whole hdf5 stack.

I am thinking now whether I should change default in py-h5py or not. Neurodamus doesn't use h5py and hence it should be fine.

ferdonline replied:
The python version currently in test uses it :/

pramodk replied:
but do they need `py-h5py+mpi` ?

ferdonline reviewed (approved):
LGTM, but would like to be sure about the `when` clause - check comment

pramodk committed:
Fix outdated sha and remove outdated version

pramodk committed:
Remove outdated version constraint; make py-h5py non-mpi by default

pramodk committed:
Address build failures:
    - spykfunc mirror fix (keep .git)
    - multisend activated in neuron for +mpi
    - py-enum34 fix for python3
    - new version of parquet converters

pramodk committed:
Fix cmake dep in pytorch and update symengine to v3

wvangeit commented on var/spack/repos/builtin/packages/py-bluepymm/package.py:
```diff
@@ -32,12 +32,12 @@ class PyBluepymm(PythonPackage):
     depends_on('py-matplotlib', type='run')
     # The below dependency should disappear once the matplotlib package is fixed
-    depends_on('py-backports-functools-lru-cache', type='run')
+    depends_on('py-backports-functools-lru-cache', type='run', when="^python@:3.3.99")
```
Just to be sure, this means 'only use when the python version is <=3.3.99 ?'

pramodk replied:
yes, thats correct.

pramodk committed:
Make py-h5py +mpi by default
